### PR TITLE
test(gno.land/sdk/vm): add unit tests for `Msg*.ValidateBasic`

### PR DESCRIPTION
**Pull request by sunspirit99**
The `vm Msg` is not yet covered by unit tests

<details><summary>Contributors' checklist...</summary>

- [ ] Added new tests, or not needed, or not feasible
- [ ] Provided an example (e.g. screenshot) to aid review or the PR is self-explanatory
- [ ] Updated the official documentation or not needed
- [ ] No breaking changes were made, or a `BREAKING CHANGE: xxx` message was included in the description
- [ ] Added references to related issues and PRs
- [ ] Provided any useful hints for running manual tests
- [ ] Added new benchmarks to [generated graphs](https://gnoland.github.io/benchmarks), if any. More info [here](https://github.com/gnolang/gno/blob/master/.benchmarks/README.md).
</details>

### CODE REVIEW

File: gno.land/pkg/sdk/vm/msg_test.go [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+
+	tests := []struct {
+		name            string
+		msg             MsgAddPackage
+		expectSignBytes string
+		expectErr       error
+	}{
+		{
+			name: "valid message",
+			msg: MsgAddPackage{
```
[the remark this answers is not in the document]
Comment (sunspirit99) [reply gno.land/pkg/sdk/vm/msg_test.go]: ef4e013fc1d17bdaa2e16d1768c86d951af03dac

File: gno.land/pkg/sdk/vm/msg_test.go [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+			if err := tc.msg.ValidateBasic(); err != nil {
+				assert.ErrorIs(t, err, tc.expectErr)
+			} else {
+				assert.Equal(t, tc.expectSignBytes, string(tc.msg.GetSignBytes()))
+			}
+		})
+	}
+}
+
+func TestMsgCall(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (sunspirit99) [reply gno.land/pkg/sdk/vm/msg_test.go]: ef4e013fc1d17bdaa2e16d1768c86d951af03dac

File: gno.land/pkg/sdk/vm/msg_test.go [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+			if err := tc.msg.ValidateBasic(); err != nil {
+				assert.ErrorIs(t, err, tc.expectErr)
+			} else {
+				assert.Equal(t, tc.expectSignBytes, string(tc.msg.GetSignBytes()))
+			}
+		})
+	}
+}
+
+func TestMsgRun(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (sunspirit99) [reply gno.land/pkg/sdk/vm/msg_test.go]: ef4e013fc1d17bdaa2e16d1768c86d951af03dac

File: gno.land/pkg/sdk/vm/msg_test.go [hunk 4]
```diff
@@ -0,0 +1,302 @@
...
+
+import (
+	"testing"
+
+	"github.com/gnolang/gno/tm2/pkg/crypto"
+	"github.com/gnolang/gno/tm2/pkg/std"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMsgAddPackage(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (sunspirit99) [reply gno.land/pkg/sdk/vm/msg_test.go]: ef4e013fc1d17bdaa2e16d1768c86d951af03dac

**Comment by ltzmaxwell** [comment]
LGTM, cross reference https://github.com/gnolang/gno/pull/2409#pullrequestreview-2242894370 for future improve, at least we can add `PkgExistError` for now.

**Review by thehowl — APPROVED** [review]

**Comment by thehowl** [comment]
> LGTM, cross reference [#2409 (review)](https://github.com/gnolang/gno/pull/2409#pullrequestreview-2242894370) for future improve, at least we can add `PkgExistError` for now.

yes; though we already test it in keeper_test.go ;)

**Pull request merged**